cmake/OpenCVDetectPython.cmake

```diff
@@ -291,6 +291,8 @@ find_python("${OPENCV_PYTHON3_VERSION}" "${MIN_VER_PYTHON3}" PYTHON3_LIBRARY PYT
     PYTHON3_INCLUDE_DIR PYTHON3_INCLUDE_DIR2 PYTHON3_PACKAGES_PATH
     PYTHON3_NUMPY_INCLUDE_DIRS PYTHON3_NUMPY_VERSION)
 
+option(PYTHON3_LIMITED_API "Build with Python Limited API" NO)
```

Comment:
Perhaps bad NumPy version handling should be moved here too somehow: https://github.com/opencv/opencv/blob/75dc334d39cba41b236116991243ae195ba85c4d/modules/python/python3/CMakeLists.txt#L5-L12
Otherwise we might get two different cmake variables (local, cache) with the same name.

Reply:
Fixed.